docs/_documentations/workingwithtemplates.md

```diff
@@ -50,3 +50,34 @@ Add your template sources to use Codewind with the framework of your choice.
    **Eclipse:** Right click on the connection in the Codewind Explorer view, and select **Manage Template Sources...**. Once the **Manage Template Sources** wizard pops up, click **Add...**.
 3. **VSCode:** Enter the URL to your template source's index file and click `Enter` to confirm.
    **Eclipse:** Fill in the fields for URL, Name, and Description. Click the **OK** button once you are finished.
+
+## Creating your own templates
+1. Choose a GitHub repository to contain all of your new template sources.
+   - Codewind uses [`codewind-templates/devfiles`](https://github.com/codewind-resources/codewind-templates/tree/master/devfiles).
+2. Within this repository, create a folder for each template source.
+   - For example, within the `devfiles` folder, Codewind has subfolders for template sources that include Go, MicroProfile, Lagom Java, and more.
+3. Each template source folder needs a `devfile.yaml` file with the following information:
+  ```
+  apiVersion: <The version of the API that you use>
+  metadata:
+  name: <The name of your template>
+  projects:
+    - name: <The name of your project>
+      source:
+        type: git
+        location: <"The GitHub URL of the template location">
+  ```
+4. In the same GitHub repository that you saved the template source folders, create an `index.json` file. In this file, add the following values for each of the templates that you want to work with in Codewind:
+  ```
+ {
+  "displayName":"<The name of your template>",
+  "description":"<The description of your template>",
+  "icon":"<The SVG graphic to be associated with your template>",
+  "language":"<The programming language that your template is written in>",
+  "projectType":"<The project type of your template>",
+  "location":"<The Git repository where the template itself is located>",
+  "links": {"self":"<The template devfile.yaml>" }
+ }
+  ```
+   - For an example `index.json` file, see the [`index-json` file](https://github.com/codewind-resources/codewind-templates/blob/master/devfiles/index.json) in the `codewind-templates` repository.
+5. Clone the [`codewind-templates` repository](https://github.com/codewind-resources/codewind-templates).
```

Comment:
I am currently unsure how to "attach" the custom templates to Codewind and the role of cloning the Codewind repository. I will ask Toby.

Reply:
you don't need to worry about the cloning of the repo, codewind takes care of that for you for any supplied template repo.  To add the templates to Codewind, that can be done through the IDE's themselves in the Template Manager.
![Screenshot 2020-04-08 at 15 01 56](https://user-images.githubusercontent.com/13151286/78793172-ef554c00-79a9-11ea-8bb5-85ae071f41ac.png)